framework/src/iotbus/iotbus_adc.c

```diff
@@ -297,7 +297,7 @@ uint32_t iotbus_adc_get_sample(iotbus_adc_context_h hnd, int timeout)
 	readsize = sizeof(struct adc_msg_s);
 
 	handle->state = IOTBUS_ADC_BUSY;
-	ret = ioctl(handle->fd, ANIOC_TRIGGER, NULL);
+	ret = ioctl(handle->fd, ANIOC_TRIGGER, 0);
 	if (ret < 0) {
 		ibdbg("[ADC] trigger error(%d)\n", ret);
 		handle->state = IOTBUS_ADC_RDY;
```

Comment:
The return value should be a negative value from iotbus_error_e.

Reply:
Thanks, I will fix it.